### 7. ¿Cuántas marcas blancas tiene mi competencia? , ¿estamos mejor o peor posicionados según precio y variedad?

* Si queremos saber cuántos productos bajo su marca blanca tiene nuestra competencia, primero debemos separar nuestro DataFrame por Supermercado. Lo hacemos a través de la columna "name", para así obtener los productos que corresponden a cada Marca Blanca.
* Luego debemos contabilizar los productos de cada marca blanca, para ello debemos filtrar los valores únicos para evitar contar artículos duplicados.

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
pd.set_option('display.float_format', lambda x: '%.2f' % x) # truco para notación cientifica
import requests as rq
import datetime

### Traemos la tabla "RAWDATA"

In [2]:
df_rawdata = pd.read_csv('raw_data.csv')
df_rawdata

category  \
0        perfumeria_e_higiene_cuidado_facial_protector_...   
1            perfumeria_e_higiene_higiene_intima_compresas   
2            perfumeria_e_higiene_higiene_intima_compresas   
3        perfumeria_e_higiene_higiene_bucal_productos_p...   
4        perfumeria_e_higiene_higiene_bucal_productos_p...   
...                                                    ...   
8223987                            fruta_y_verdura_verdura   
8223988                                      bebidas_aguas   
8223989                            fruta_y_verdura_verdura   
8223990                              fruta_y_verdura_fruta   
8223991                            fruta_y_verdura_verdura   

                                                      name description  price  \
0           LIPOSAN protector labial soft rose envase 1 ud         NaN   2.39   
1               AUSONIA compresas normal alas bolsa 14 uds         NaN   2.49   
2        EVAX Cottonlike compresa super plus con alas b...         NaN   2.79   
3        POLIDENT tabletas limpiadoras para prótesis de...         NaN   7.95   
4        IWHITE kit de blanquamiento dental profesional...         NaN  36.99   
...                                                    ...         ...    ...   
8223987                                    Tomate ensalada     Bandeja   2.09   
8223988             DIA agua mineral natural botella 50 cl         NaN   0.22   
8223989                                             Pepino       Pieza   0.38   
8223990                                    Nectarinas mini     Bandeja   2.19   
8223991                                         Zanahorias     Paquete   0.59   

         reference_price reference_unit          insert_date  \
0                   2.39             ud  2021-03-06 00:00:00   
1                   0.18             ud  2021-03-06 00:00:00   
2                   0.23             ud  2021-03-06 00:00:00   
3                   0.11             ud  2021-03-06 00:00:00   
4                   3.70             ud  2021-03-06 00:00:00   
...                  ...            ...                  ...   
8223987             2.09             kg  2022-09-15 00:00:00   
8223988             0.44              l  2022-09-15 00:00:00   
8223989             1.89             kg  2022-09-15 00:00:00   
8223990             2.19             kg  2022-09-15 00:00:00   
8223991             1.18             kg  2022-09-15 00:00:00   

                               product_id id_reference_supermarket  
0        45d83ce427cf60c02e1e6f3d28f58ed6                       M3  
1        3fa10738dcbe5f09646eaf39a46a2871                       M3  
2        5f5bc0a466375a16d01a6aed6c66e413                       M3  
3        0a1e56be01f8766d0bb87f3e42fd2abf                       M3  
4        b150c5e27da5c91d13a5b570c9b3efd3                       M3  
...                                   ...                      ...  
8223987  83dca36ff54e06879a2d14d2f4120859                       M1  
8223988  34d7b9f9d64d3579ed800d5b7f2db8c6                       M3  
8223989  e390924043f9e289d1db5d60f24a98df                       M1  
8223990  57bff22438b12272070fc553423dc6ee                       M1  
8223991  c0ac3da899e51308db2492c91aebbcb2                       M1  

[8223992 rows x 9 columns]

### Traemos la tabla "Competidores"

In [3]:
df_competidores = pd.read_csv('Super_competitors .csv')
df_competidores

ID       Name
0  M1  Mercadona
1  M2  Carrefour
2  M3        Dia

### Usamos la función Merge para unir las tablas "competidores" y "rawdata"

In [4]:
## Join de tablas competidores y parte de rawdata
df_data = pd.merge(df_rawdata, df_competidores,how = 'left', left_on = 'id_reference_supermarket', right_on = 'ID')
display(df_data)

category  \
0        perfumeria_e_higiene_cuidado_facial_protector_...   
1            perfumeria_e_higiene_higiene_intima_compresas   
2            perfumeria_e_higiene_higiene_intima_compresas   
3        perfumeria_e_higiene_higiene_bucal_productos_p...   
4        perfumeria_e_higiene_higiene_bucal_productos_p...   
...                                                    ...   
8223987                            fruta_y_verdura_verdura   
8223988                                      bebidas_aguas   
8223989                            fruta_y_verdura_verdura   
8223990                              fruta_y_verdura_fruta   
8223991                            fruta_y_verdura_verdura   

                                                      name description  price  \
0           LIPOSAN protector labial soft rose envase 1 ud         NaN   2.39   
1               AUSONIA compresas normal alas bolsa 14 uds         NaN   2.49   
2        EVAX Cottonlike compresa super plus con alas b...         NaN   2.79   
3        POLIDENT tabletas limpiadoras para prótesis de...         NaN   7.95   
4        IWHITE kit de blanquamiento dental profesional...         NaN  36.99   
...                                                    ...         ...    ...   
8223987                                    Tomate ensalada     Bandeja   2.09   
8223988             DIA agua mineral natural botella 50 cl         NaN   0.22   
8223989                                             Pepino       Pieza   0.38   
8223990                                    Nectarinas mini     Bandeja   2.19   
8223991                                         Zanahorias     Paquete   0.59   

         reference_price reference_unit          insert_date  \
0                   2.39             ud  2021-03-06 00:00:00   
1                   0.18             ud  2021-03-06 00:00:00   
2                   0.23             ud  2021-03-06 00:00:00   
3                   0.11             ud  2021-03-06 00:00:00   
4                   3.70             ud  2021-03-06 00:00:00   
...                  ...            ...                  ...   
8223987             2.09             kg  2022-09-15 00:00:00   
8223988             0.44              l  2022-09-15 00:00:00   
8223989             1.89             kg  2022-09-15 00:00:00   
8223990             2.19             kg  2022-09-15 00:00:00   
8223991             1.18             kg  2022-09-15 00:00:00   

                               product_id id_reference_supermarket  ID  \
0        45d83ce427cf60c02e1e6f3d28f58ed6                       M3  M3   
1        3fa10738dcbe5f09646eaf39a46a2871                       M3  M3   
2        5f5bc0a466375a16d01a6aed6c66e413                       M3  M3   
3        0a1e56be01f8766d0bb87f3e42fd2abf                       M3  M3   
4        b150c5e27da5c91d13a5b570c9b3efd3                       M3  M3   
...                                   ...                      ...  ..   
8223987  83dca36ff54e06879a2d14d2f4120859                       M1  M1   
8223988  34d7b9f9d64d3579ed800d5b7f2db8c6                       M3  M3   
8223989  e390924043f9e289d1db5d60f24a98df                       M1  M1   
8223990  57bff22438b12272070fc553423dc6ee                       M1  M1   
8223991  c0ac3da899e51308db2492c91aebbcb2                       M1  M1   

              Name  
0              Dia  
1              Dia  
2              Dia  
3              Dia  
4              Dia  
...            ...  
8223987  Mercadona  
8223988        Dia  
8223989  Mercadona  
8223990  Mercadona  
8223991  Mercadona  

[8223992 rows x 11 columns]

### Creamos la tabla "productos"

Con el fin de reducir el dataset a la cantidad de columnas necesarias para nuestro análisis.

In [5]:
df_productos = df_data[['Name','category','name','price','insert_date','product_id']]
df_productos

Name                                           category  \
0              Dia  perfumeria_e_higiene_cuidado_facial_protector_...   
1              Dia      perfumeria_e_higiene_higiene_intima_compresas   
2              Dia      perfumeria_e_higiene_higiene_intima_compresas   
3              Dia  perfumeria_e_higiene_higiene_bucal_productos_p...   
4              Dia  perfumeria_e_higiene_higiene_bucal_productos_p...   
...            ...                                                ...   
8223987  Mercadona                            fruta_y_verdura_verdura   
8223988        Dia                                      bebidas_aguas   
8223989  Mercadona                            fruta_y_verdura_verdura   
8223990  Mercadona                              fruta_y_verdura_fruta   
8223991  Mercadona                            fruta_y_verdura_verdura   

                                                      name  price  \
0           LIPOSAN protector labial soft rose envase 1 ud   2.39   
1               AUSONIA compresas normal alas bolsa 14 uds   2.49   
2        EVAX Cottonlike compresa super plus con alas b...   2.79   
3        POLIDENT tabletas limpiadoras para prótesis de...   7.95   
4        IWHITE kit de blanquamiento dental profesional...  36.99   
...                                                    ...    ...   
8223987                                    Tomate ensalada   2.09   
8223988             DIA agua mineral natural botella 50 cl   0.22   
8223989                                             Pepino   0.38   
8223990                                    Nectarinas mini   2.19   
8223991                                         Zanahorias   0.59   

                 insert_date                        product_id  
0        2021-03-06 00:00:00  45d83ce427cf60c02e1e6f3d28f58ed6  
1        2021-03-06 00:00:00  3fa10738dcbe5f09646eaf39a46a2871  
2        2021-03-06 00:00:00  5f5bc0a466375a16d01a6aed6c66e413  
3        2021-03-06 00:00:00  0a1e56be01f8766d0bb87f3e42fd2abf  
4        2021-03-06 00:00:00  b150c5e27da5c91d13a5b570c9b3efd3  
...                      ...                               ...  
8223987  2022-09-15 00:00:00  83dca36ff54e06879a2d14d2f4120859  
8223988  2022-09-15 00:00:00  34d7b9f9d64d3579ed800d5b7f2db8c6  
8223989  2022-09-15 00:00:00  e390924043f9e289d1db5d60f24a98df  
8223990  2022-09-15 00:00:00  57bff22438b12272070fc553423dc6ee  
8223991  2022-09-15 00:00:00  c0ac3da899e51308db2492c91aebbcb2  

[8223992 rows x 6 columns]

In [6]:
df_productos.dtypes

Name            object
category        object
name            object
price          float64
insert_date     object
product_id      object
dtype: object

#### Debemos cambiar el tipo de dato de la columna "insert_date"

La columna que contiene las fechas no es del tipo adecuado, por lo que debemos reconvertirla en un formato adecuado.

In [7]:
df_productos['insert_date'] = pd.to_datetime(df_productos['insert_date'])
df_productos.dtypes
## Esta función no toma las horas, solo fecha

C:\Users\Usuario\AppData\Local\Temp\ipykernel_9960\4231573947.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_productos['insert_date'] = pd.to_datetime(df_productos['insert_date'])


Name                   object
category               object
name                   object
price                 float64
insert_date    datetime64[ns]
product_id             object
dtype: object

#### *Al separar la información por supermercado, podemos filtrar luego los productos pertenecientes a las marcas blancas de los mismos.*

### Supermercados Día

Buscamos filtar aquellos productos pertenecientes al Supermercado Día:

In [8]:
solodia = df_productos['Name'] == 'Dia'
dia = df_productos[solodia]
dia

Name                                           category  \
0        Dia  perfumeria_e_higiene_cuidado_facial_protector_...   
1        Dia      perfumeria_e_higiene_higiene_intima_compresas   
2        Dia      perfumeria_e_higiene_higiene_intima_compresas   
3        Dia  perfumeria_e_higiene_higiene_bucal_productos_p...   
4        Dia  perfumeria_e_higiene_higiene_bucal_productos_p...   
...      ...                                                ...   
8223869  Dia  cuidado_personal_bano_e_higiene_personal_desod...   
8223870  Dia  cuidado_personal_bano_e_higiene_personal_desod...   
8223871  Dia  cuidado_personal_bano_e_higiene_personal_desod...   
8223872  Dia  cuidado_personal_bano_e_higiene_personal_desod...   
8223988  Dia                                      bebidas_aguas   

                                                      name  price insert_date  \
0           LIPOSAN protector labial soft rose envase 1 ud   2.39  2021-03-06   
1               AUSONIA compresas normal alas bolsa 14 uds   2.49  2021-03-06   
2        EVAX Cottonlike compresa super plus con alas b...   2.79  2021-03-06   
3        POLIDENT tabletas limpiadoras para prótesis de...   7.95  2021-03-06   
4        IWHITE kit de blanquamiento dental profesional...  36.99  2021-03-06   
...                                                    ...    ...         ...   
8223869    TULIPAN NEGRO desodorante original spray 200 ml   2.51  2022-09-15   
8223870            DOVE desodorante original roll on 50 ml   2.49  2022-09-15   
8223871  DIA IMAQE desodorante con extracto de rosa mos...   0.75  2022-09-15   
8223872  DIA IMAQE desodorante para pieles atópicas rol...   0.75  2022-09-15   
8223988             DIA agua mineral natural botella 50 cl   0.22  2022-09-15   

                               product_id  
0        45d83ce427cf60c02e1e6f3d28f58ed6  
1        3fa10738dcbe5f09646eaf39a46a2871  
2        5f5bc0a466375a16d01a6aed6c66e413  
3        0a1e56be01f8766d0bb87f3e42fd2abf  
4        b150c5e27da5c91d13a5b570c9b3efd3  
...                                   ...  
8223869  0162dba2776ea18ebe09705c07429fd7  
8223870  ad71bf9c32b17adcc6d88318de168283  
8223871  1a3192873d61a6cc441cfabb488872bf  
8223872  458ce35de717f4123ccbbdb5d98a1ec4  
8223988  34d7b9f9d64d3579ed800d5b7f2db8c6  

[3200969 rows x 6 columns]

Filtramos aquellos productos que se encuentran bajo la marca blanca de Día

In [10]:
blancadia=df_productos[df_productos['name'].str.contains('DIA', na=False)]
blancadia

Name                                           category  \
38       Dia  drogueria_y_limpieza_celulosa_servilletas_de_p...   
39       Dia  drogueria_y_limpieza_celulosa_servilletas_de_p...   
40       Dia  drogueria_y_limpieza_celulosa_servilletas_de_p...   
41       Dia  drogueria_y_limpieza_celulosa_servilletas_de_p...   
42       Dia  drogueria_y_limpieza_celulosa_servilletas_de_p...   
...      ...                                                ...   
8223779  Dia                    congelados_pizzas_bases_y_masas   
8223781  Dia                    congelados_pizzas_bases_y_masas   
8223871  Dia  cuidado_personal_bano_e_higiene_personal_desod...   
8223872  Dia  cuidado_personal_bano_e_higiene_personal_desod...   
8223988  Dia                                      bebidas_aguas   

                                                      name  price insert_date  \
38          DIA servilletas blancas 1 capa paquete 200 uds   0.89  2021-03-06   
39       DIA servilletas decoradas 2 capas paquete 100 uds   1.65  2021-03-06   
40       DIA servilletas negras cocktail 2 capas 22x22 ...   1.15  2021-03-06   
41       DIA servilletas rosadas 2 capas 33x33 cm paque...   1.09  2021-03-06   
42       DIA servilletas color 2 capas 33x33 cm paquete...   1.10  2021-03-06   
...                                                    ...    ...         ...   
8223779                    DIA pizza campestre caja 400 gr   1.99  2022-09-15   
8223781          DIA mini pizzas jamón y queso caja 270 gr   2.05  2022-09-15   
8223871  DIA IMAQE desodorante con extracto de rosa mos...   0.75  2022-09-15   
8223872  DIA IMAQE desodorante para pieles atópicas rol...   0.75  2022-09-15   
8223988             DIA agua mineral natural botella 50 cl   0.22  2022-09-15   

                               product_id  
38       4a8b9fafdc1905275c85f5b602c35ce6  
39       d86d99be6ef14de523ce24e93176ca89  
40       c0da5ae927d3276bea4ff275fedc5086  
41       278ce4de71a9c146875dfe5c91a6fffe  
42       ead1441643f2580f65f022b362662684  
...                                   ...  
8223779  6dfbf932ec850e92338c929dc7b936af  
8223781  b0a8232eef7b77a8eabea397e1e27e68  
8223871  1a3192873d61a6cc441cfabb488872bf  
8223872  458ce35de717f4123ccbbdb5d98a1ec4  
8223988  34d7b9f9d64d3579ed800d5b7f2db8c6  

[783543 rows x 6 columns]

 *De los 3200969 productos (obtenidos en el dataframe dia), hay un total de 783543 registros de su marca blanca.*
 *En este caso, estamos viendo productos repetidos. Ya que estarían los id de cada producto que se generan todos los días de cada año, repitiéndose el mismo producto. Debemos hallar valores únicos.*

#### Valores únicos para Dia

In [11]:
uniqdia = pd.DataFrame(blancadia['product_id'].unique())
uniqdia

0
0     4a8b9fafdc1905275c85f5b602c35ce6
1     d86d99be6ef14de523ce24e93176ca89
2     c0da5ae927d3276bea4ff275fedc5086
3     278ce4de71a9c146875dfe5c91a6fffe
4     ead1441643f2580f65f022b362662684
...                                ...
2409  aaf5d368b6e70c219b671aa12c9cddee
2410  fc6fa7743b9bbf1032280d5e438a726a
2411  0a9fb4ab1497a6b49b5873f3e94d9cca
2412  54380f2d7940608e20f6d0bc3e0e038a
2413  40105bc24be74e71a781052ecc4bdaa3

[2414 rows x 1 columns]

#### *Existen 2414 productos únicos de marcas blancas pertenecientes a Dia*

In [12]:
blancadia.describe()

price
count 783539.00
mean       2.03
std        1.90
min        0.08
25%        1.00
50%        1.49
75%        2.25
max       23.50

### Supermercados Carrefour

Buscamos filtar aquellos productos pertenecientes al Supermercado Carrefour:

In [45]:
solocarre = df_productos['Name'] == 'Carrefour'
carrefour = df_productos[solocarre]
carrefour

Name                                           category  \
2053     Carrefour                               bebidas_vinos_rosado   
2054     Carrefour                               bebidas_vinos_blanco   
2055     Carrefour                              bebidas_alcoholes_ron   
2056     Carrefour                               bebidas_vinos_blanco   
2057     Carrefour        bebidas_aguas_y_zumos_agua_hasta_075_litros   
...            ...                                                ...   
8223977  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_afe...   
8223978  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_afe...   
8223979  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_aft...   
8223980  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_aft...   
8223981  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_aft...   

                                                      name  price insert_date  \
2053                                      Muga Rosado 2019   9.15  2021-03-06   
2054                        Blanc Pescador Espumoso Blanco   3.99  2021-03-06   
2055                      Ron Brugal añejo superior 70 cl.  13.80  2021-03-06   
2056                                          Arabe Blanco   3.99  2021-03-06   
2057            Agua mineral Carrefour Kids natural 33 cl.   0.19  2021-03-06   
...                                                    ...    ...         ...   
8223977      Espuma de afeitar sensitive Nivea Men 200 ml.   3.05  2022-09-15   
8223978         Gel de afeitar Sensitive Nivea Men 200 ml.   3.79  2022-09-15   
8223979    After shave bálsamo Sensitive Nivea Men 100 ml.   4.75  2022-09-15   
8223980        After shave bálsamo Classic La Toja 100 ml.   5.45  2022-09-15   
8223981  After shave bálsamo extra sensible La Toja 100...   5.45  2022-09-15   

                               product_id  
2053     4a9bb57a9ebbad18dff028c702d70c0e  
2054     e2edfbb48927c755131eab0016995453  
2055     e86bfe04af8fdab975a3ac9a86276e3f  
2056     b0077d68ba056d38d0ac11b61909cdd7  
2057     8d83f3fd26260b30a056ba9aba86cd8a  
...                                   ...  
8223977  0db25207ed2e73d839e0ead0c34d8cac  
8223978  78fd7919b63b2f4c5d955c87b0939835  
8223979  e2ac225b7436cf2da4f53e99866316bb  
8223980  2bd2f1b34a3bad85779fe909ea3a0e66  
8223981  f017823262284f0ea6fd6343ced93b49  

[1693583 rows x 6 columns]

Filtramos aquellos productos que se encuentran bajo la marca blanca de Carrefour

In [13]:
blancacar=df_productos[df_productos['name'].str.contains('Carrefour', na=False)]
blancacar

Name                                           category  \
2057     Carrefour        bebidas_aguas_y_zumos_agua_hasta_075_litros   
2059     Carrefour           bebidas_aguas_y_zumos_zumos_refrigerados   
2076     Carrefour           bebidas_aguas_y_zumos_zumos_refrigerados   
2088     Carrefour                              bebidas_cerveza_negra   
2115     Carrefour                         bebidas_alcoholes_vermouth   
...            ...                                                ...   
8223962  Carrefour  perfumeria_e_higiene_boca_y_sonrisa_protector_...   
8223963  Carrefour  perfumeria_e_higiene_boca_y_sonrisa_protector_...   
8223973  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_dep...   
8223974  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_dep...   
8223975  Carrefour  perfumeria_e_higiene_depilacion_y_afeitado_afe...   

                                                      name  price insert_date  \
2057            Agua mineral Carrefour Kids natural 33 cl.   0.19  2021-03-06   
2059     Zumo de piña y mango Carrefour Selección expri...   2.05  2021-03-06   
2076     Zumo de uva roja, granada y grosella Carrefour...   1.55  2021-03-06   
2088                Cerveza negra Carrefour botella 33 cl.   0.65  2021-03-06   
2115             Vermut Carrefour Monte Esquinza rojo 1 l.   3.11  2021-03-06   
...                                                    ...    ...         ...   
8223962                     Bálsamo labial Carrefour 2 ud.   2.39  2022-09-15   
8223963  Vaselina para labios perfumada frambuesa Soft ...   1.60  2022-09-15   
8223973  Bandas de cera corporales depilatoria frías Ca...   1.55  2022-09-15   
8223974  Gel depilatorio piel sensible con manteca de k...   1.99  2022-09-15   
8223975  Espuma de afeitar piel sensible Carrefour Men ...   1.18  2022-09-15   

                               product_id  
2057     8d83f3fd26260b30a056ba9aba86cd8a  
2059     adfe3f4d2b7757a578b34cc21185dd47  
2076     b371c0878eb4c3525ef1c74aef750669  
2088     8e71e0b94a483137019b1782441eca4f  
2115     49a0d94bbfaeee74db5474aa02e96da4  
...                                   ...  
8223962  a984257dc8cb0a58709407702f2a0429  
8223963  86f782ad7b5cd8638f162bcc50c029df  
8223973  a281261f66deb07fce2d8ec9550d7d5e  
8223974  a8c45307bc56b483e890932ffb4ebb60  
8223975  a0137ae747a7ac5e2f606b9956d93f50  

[512024 rows x 6 columns]

 *De los 1693583 productos (obtenidos en el dataframe carrefour), hay un total de 512024 registros de su marca blanca.*
 *En este caso, estamos viendo productos repetidos, por lo que debemos hallar valores únicos*

In [14]:
uniqcarrefour = pd.DataFrame(blancacar['product_id'].unique())
uniqcarrefour

0
0     8d83f3fd26260b30a056ba9aba86cd8a
1     adfe3f4d2b7757a578b34cc21185dd47
2     b371c0878eb4c3525ef1c74aef750669
3     8e71e0b94a483137019b1782441eca4f
4     49a0d94bbfaeee74db5474aa02e96da4
...                                ...
3052  8b140e80ab33989d668914866ba12c4c
3053  75e69ea184db6a0884d5a35398093ea2
3054  a0086353c2fa6005acbd826278a22166
3055  c729198f6c71a8b4d0fb4b8827093874
3056  0b4489f585b44df5e77a7ff3dbec750e

[3057 rows x 1 columns]

#### *Existen 3057 productos únicos de marcas blancas pertenecientes a Carrefour*

In [15]:
blancacar.describe()

price
count 512024.00
mean       2.48
std        5.65
min        0.19
25%        0.99
50%        1.60
75%        2.50
max      149.90

### Supermercados Mercadona

* En el caso particular de Mercadona, sus marcas blancas no se encuentran bajo el mismo nombre. Por lo que primero debemos filtrar los mismos para obtener el total de cada una.
* Las marcas blancas de Mercadona son: Hacendado, Bosque Verde, Deliplus y Steinburg.

Buscamos filtar aquellos productos pertenecientes al Supermercado Mercadona:

In [17]:
solomercadona = df_productos['Name'] == 'Mercadona'
mercadona = df_productos[solomercadona]
mercadona

Name                                 category  \
1055     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1056     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1057     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1058     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1059     Mercadona  agua_y_refrescos_isotonico_y_energetico   
...            ...                                      ...   
8223986  Mercadona                    fruta_y_verdura_fruta   
8223987  Mercadona                  fruta_y_verdura_verdura   
8223989  Mercadona                  fruta_y_verdura_verdura   
8223990  Mercadona                    fruta_y_verdura_fruta   
8223991  Mercadona                  fruta_y_verdura_verdura   

                                                    name  price insert_date  \
1055     Bebida para deportistas sabor cítrico Hacendado   0.85  2021-03-06   
1056     Bebida para deportistas sabor cítrico Hacendado   1.92  2021-03-06   
1057     Bebida para deportistas sabor cítrico Hacendado   0.32  2021-03-06   
1058                   Bebida isotónica naranja Aquarius   1.55  2021-03-06   
1059                   Bebida isotónica naranja Aquarius   5.92  2021-03-06   
...                                                  ...    ...         ...   
8223986                                          Granada   0.62  2022-09-15   
8223987                                  Tomate ensalada   2.09  2022-09-15   
8223989                                           Pepino   0.38  2022-09-15   
8223990                                  Nectarinas mini   2.19  2022-09-15   
8223991                                       Zanahorias   0.59  2022-09-15   

                               product_id  
1055     cc6af45cd819ddd9c69c7a574b840e29  
1056     f6a62a8c304a0e818f869aa2ea02df8a  
1057     90361e7447fa2d5d4ea757b57d4e6a64  
1058     ed6b157243bb5b81f11dccd5b5d394e9  
1059     727c0bb06307a6fe756835b1d0eea3e6  
...                                   ...  
8223986  5d5c39d669d3dcac2549932b2d4790c4  
8223987  83dca36ff54e06879a2d14d2f4120859  
8223989  e390924043f9e289d1db5d60f24a98df  
8223990  57bff22438b12272070fc553423dc6ee  
8223991  c0ac3da899e51308db2492c91aebbcb2  

[3329440 rows x 6 columns]

Ahora filtramos por cada marca blanca de Mercadona:

#### Hacendado

In [18]:
blancahac=df_productos[df_productos['name'].str.contains('Hacendado', na=False)]
blancahac

Name                                 category  \
1055     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1056     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1057     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1068     Mercadona  agua_y_refrescos_isotonico_y_energetico   
1093     Mercadona        agua_y_refrescos_refresco_de_cola   
...            ...                                      ...   
8223840  Mercadona             zumos_tomate_y_otros_sabores   
8223841  Mercadona             zumos_tomate_y_otros_sabores   
8223842  Mercadona             zumos_tomate_y_otros_sabores   
8223843  Mercadona             zumos_tomate_y_otros_sabores   
8223844  Mercadona             zumos_tomate_y_otros_sabores   

                                                      name  price insert_date  \
1055       Bebida para deportistas sabor cítrico Hacendado   0.85  2021-03-06   
1056       Bebida para deportistas sabor cítrico Hacendado   1.92  2021-03-06   
1057       Bebida para deportistas sabor cítrico Hacendado   0.32  2021-03-06   
1068                     Bebida energética light Hacendado   0.39  2021-03-06   
1093                               Refresco cola Hacendado   0.53  2021-03-06   
...                                                    ...    ...         ...   
8223840                          Zumo de manzana Hacendado   0.65  2022-09-15   
8223841  Néctar de maracuyá Hacendado fruta de la pasió...   0.90  2022-09-15   
8223842   Bebida de pomelo Hacendado sin azúcares añadidos   1.10  2022-09-15   
8223843     Bebida de pera Hacendado sin azúcares añadidos   1.00  2022-09-15   
8223844                Zumo de manzana selección Hacendado   1.40  2022-09-15   

                               product_id  
1055     cc6af45cd819ddd9c69c7a574b840e29  
1056     f6a62a8c304a0e818f869aa2ea02df8a  
1057     90361e7447fa2d5d4ea757b57d4e6a64  
1068     7ed879b482befb1385cbd68053bd5cdc  
1093     0fee679e6cd7841821c432b1fa529ccc  
...                                   ...  
8223840  8471d4a9d632274d2ce253cec4830296  
8223841  6a05bacc0c7d7bffdc2cec006b1dd12b  
8223842  d60df813f3282eafe303e5d34c83b888  
8223843  c68f9a71361902292a66ce815861a0e3  
8223844  e754b674fe121395c8ea78dc91bb2d9a  

[1127883 rows x 6 columns]

 *De los 3329440 productos (obtenidos en el dataframe mercadona), hay un total de 1127883 registros que se corresponden con Hacendado.*
 *Debemos tener en cuenta que  estamos viendo productos repetidos, por lo que debemos hallar valores únicos.*

In [19]:
uniqhac = pd.DataFrame(blancahac['product_id'].unique())
uniqhac

0
0     cc6af45cd819ddd9c69c7a574b840e29
1     f6a62a8c304a0e818f869aa2ea02df8a
2     90361e7447fa2d5d4ea757b57d4e6a64
3     7ed879b482befb1385cbd68053bd5cdc
4     0fee679e6cd7841821c432b1fa529ccc
...                                ...
2741  115588048e3d76e8acf769fa3d498bff
2742  de16e13dcc1e583b2aeebfefcd979233
2743  670f8dfe00f817b45d282312d40d290c
2744  c391fef74e016a05a15268784795c118
2745  0d09fe77827a342b9f0de6fec840f1df

[2746 rows x 1 columns]

#### *Hacendado de Mercadona tiene 2746 productos diferentes*

In [20]:
blancahac.describe()

price
count 1127883.00
mean        2.01
std         2.43
min         0.12
25%         1.00
50%         1.45
75%         2.15
max        79.95

#### Bosque Verde

In [21]:
blancabos=df_productos[df_productos['name'].str.contains('Bosque Verde', na=False)]
blancabos

Name                                       category  \
1207     Mercadona  limpieza_y_hogar_limpieza_muebles_y_multiusos   
1261     Mercadona  limpieza_y_hogar_limpieza_muebles_y_multiusos   
1278     Mercadona  limpieza_y_hogar_limpieza_muebles_y_multiusos   
1753     Mercadona  limpieza_y_hogar_detergente_y_suavizante_ropa   
1754     Mercadona  limpieza_y_hogar_detergente_y_suavizante_ropa   
...            ...                                            ...   
8223426  Mercadona      limpieza_y_hogar_pilas_y_bolsas_de_basura   
8223427  Mercadona      limpieza_y_hogar_pilas_y_bolsas_de_basura   
8223428  Mercadona      limpieza_y_hogar_pilas_y_bolsas_de_basura   
8223429  Mercadona      limpieza_y_hogar_pilas_y_bolsas_de_basura   
8223431  Mercadona      limpieza_y_hogar_pilas_y_bolsas_de_basura   

                                                      name  price insert_date  \
1207              Limpiador muebles y parquet Bosque Verde   1.95  2021-03-06   
1261     Limpia tapicerías y alfombras con espuma activ...   1.58  2021-03-06   
1278            Limpia tapicerías espuma seca Bosque Verde   2.05  2021-03-06   
1753        Suavizante ropa talco Bosque Verde concentrado   1.75  2021-03-06   
1754         Suavizante ropa rosa Bosque Verde concentrado   1.75  2021-03-06   
...                                                    ...    ...         ...   
8223426      Bolsa de basura Bosque Verde 30L cubo mediano   1.70  2022-09-15   
8223427  Bolsa de basura saco Bosque Verde 120L cubo in...   1.25  2022-09-15   
8223428  Bolsa de basura mini Bosque Verde 10L cubo peq...   1.35  2022-09-15   
8223429  Bolsa de basura cubo alto Bosque Verde 30L cub...   1.90  2022-09-15   
8223431  Bolsa de basura mini Bosque Verde 5L cubo pequeño   1.25  2022-09-15   

                               product_id  
1207     ed76770e4ef6a331a0801bafbf15ac45  
1261     59398592319bbeb29549a404e7ef97aa  
1278     74c5f0374853b03adca6a9667d43c678  
1753     d4cd1f4188141209dd147d29769b5575  
1754     1f72601529303bf32075968f496168a2  
...                                   ...  
8223426  cec8503c41cd28c6841cc30f91dc2c96  
8223427  bcaedfe0a4adbdaaa2824d60c1f173f8  
8223428  438a94b812aba99fca972cb5cb1b9073  
8223429  b910035dd5e2f6c8c36f4604aeff03b7  
8223431  51f3a385a1724b814bfeac27f120f709  

[233975 rows x 6 columns]

 *De los 3329440 productos (obtenidos en el dataframe mercadona), hay un total de 233975 registros que se corresponden con Bosque Verde.*
 *Debemos tener en cuenta que  estamos viendo productos repetidos, por lo que debemos hallar valores únicos.*

In [22]:
uniqbos = pd.DataFrame(blancabos['product_id'].unique())
uniqbos

0
0    ed76770e4ef6a331a0801bafbf15ac45
1    59398592319bbeb29549a404e7ef97aa
2    74c5f0374853b03adca6a9667d43c678
3    d4cd1f4188141209dd147d29769b5575
4    1f72601529303bf32075968f496168a2
..                                ...
527  7b5f433db5395589b028150aa0b8b2bf
528  413eaf590626646a6e4a7decd9416bba
529  5c1b20bdfd7ead9ecce363aad61bcc4d
530  7ebb69b88dda056dd9ef6016c5d59e11
531  5620c9176cce195e1e11aa7b7c6c5dd8

[532 rows x 1 columns]

#### *Bosque Verde de Mercadona tiene 532 productos diferentes.* 

In [23]:
blancabos.describe()

price
count 233975.00
mean       2.02
std        1.37
min        0.55
25%        1.20
50%        1.70
75%        2.35
max       16.45

#### Deliplus

In [24]:
blancadel=df_productos[df_productos['name'].str.contains('Deliplus', na=False)]
blancadel

Name                       category  \
1236     Mercadona  bebe_biberon_chupete_y_menaje   
11584    Mercadona              maquillaje_labios   
11585    Mercadona              maquillaje_labios   
11586    Mercadona   maquillaje_colorete_y_polvos   
11587    Mercadona   maquillaje_colorete_y_polvos   
...            ...                            ...   
8223525  Mercadona  maquillaje_pinceles_y_brochas   
8223526  Mercadona  maquillaje_pinceles_y_brochas   
8223527  Mercadona  maquillaje_pinceles_y_brochas   
8223528  Mercadona  maquillaje_pinceles_y_brochas   
8223529  Mercadona  maquillaje_pinceles_y_brochas   

                                                      name  price insert_date  \
1236      Limpiador bebé para biberones y tetinas Deliplus   2.60  2021-03-06   
11584       Vaselina perfumada para labios Deliplus neutra   1.60  2021-03-06   
11585    Vaselina perfumada para labios Deliplus frambuesa   1.60  2021-03-06   
11586             Polvo compacto balm me Deliplus 02 siena   6.00  2021-03-06   
11587             Polvo compacto balm me Deliplus 01 arena   6.00  2021-03-06   
...                                                    ...    ...         ...   
8223525                         Pincel delineador Deliplus   5.00  2022-09-15   
8223526                         Pincel iluminador Deliplus   6.00  2022-09-15   
8223527                             Pincel polvos Deliplus   6.00  2022-09-15   
8223528                           Pincel contorno Deliplus   6.00  2022-09-15   
8223529                         Brocha maquillaje Deliplus   5.25  2022-09-15   

                               product_id  
1236     4e5d6a815724e879eff9d015fc1092d7  
11584    4e53340e8fa8740ce78ad7a426c06029  
11585    dc269c0433b0df5e24fddec8a8f6f36e  
11586    186dca76f2dbd3ffb6689462cbcd83f0  
11587    59d88f5b8d8f1b136427643e5a33696d  
...                                   ...  
8223525  0cd356079da73d1df6e05a304a2f033f  
8223526  0fd79d5e481d038856fca0b70a9e831b  
8223527  47ed1940fcaf480b78412da919138010  
8223528  c4b5d53eb721f4e2c8bea3bd645adf60  
8223529  8f05d2aa802476da2734ff3d6a6d53d2  

[522778 rows x 6 columns]

 *De los 3329440 productos (obtenidos en el dataframe mercadona), hay un total de 522778 registros que se corresponden con Deliplus.*
 *Debemos tener en cuenta que  estamos viendo productos repetidos, por lo que debemos hallar valores únicos.*

In [25]:
uniqdel = pd.DataFrame(blancadel['product_id'].unique())
uniqdel

0
0     4e5d6a815724e879eff9d015fc1092d7
1     4e53340e8fa8740ce78ad7a426c06029
2     dc269c0433b0df5e24fddec8a8f6f36e
3     186dca76f2dbd3ffb6689462cbcd83f0
4     59d88f5b8d8f1b136427643e5a33696d
...                                ...
1603  f9c6dea6636626a40813b35e62f58596
1604  f99bc82b885ebc1955dcb805d2945c74
1605  a5548af9ecd528cbd8dd626e19658897
1606  b1e8118a99663fc09be466e7bf790e97
1607  894577cc775900ebeddaa855bdb55fe0

[1608 rows x 1 columns]

#### *Deliplus de Mercadona tiene 1608 productos diferentes.*

In [26]:
blancadel.describe()

price
count 522778.00
mean       3.19
std        1.91
min        0.50
25%        1.70
50%        2.95
75%        4.00
max       17.55

#### Steinburg

In [27]:
blancaste=df_productos[df_productos['name'].str.contains('Steinburg', na=False)]
blancaste

Name        category  \
1237     Mercadona  bodega_cerveza   
1238     Mercadona  bodega_cerveza   
1247     Mercadona  bodega_cerveza   
1248     Mercadona  bodega_cerveza   
1249     Mercadona  bodega_cerveza   
...            ...             ...   
8221500  Mercadona  bodega_cerveza   
8221505  Mercadona  bodega_cerveza   
8221506  Mercadona  bodega_cerveza   
8221509  Mercadona  bodega_cerveza   
8221510  Mercadona  bodega_cerveza   

                                                      name  price insert_date  \
1237                             Cerveza Clásica Steinburg   2.76  2021-03-06   
1238                             Cerveza Clásica Steinburg   0.23  2021-03-06   
1247                               Cerveza Suave Steinburg   2.64  2021-03-06   
1248                               Cerveza Suave Steinburg   0.22  2021-03-06   
1249                            Cerveza Especial Steinburg   3.36  2021-03-06   
...                                                    ...    ...         ...   
8221500                         Cerveza Especial Steinburg   0.37  2022-09-15   
8221505  Cerveza Radler con zumo natural de limón Stein...   0.38  2022-09-15   
8221506  Cerveza Radler Steinburg con zumo natural de l...   0.58  2022-09-15   
8221509               Cerveza Shandy sabor limón Steinburg   0.35  2022-09-15   
8221510               Cerveza Shandy Steinburg sabor limón   0.53  2022-09-15   

                               product_id  
1237     3c6d92df67a1e42852133818f6321b09  
1238     5d0c9d51dc38b9c139b22308cf50d1f8  
1247     2318cf2ee797dc035e4846a270ef9cf5  
1248     2e181ce72e732e6e18cdeef3be8921b3  
1249     d239934fa26e6a34473f1b2f324096a1  
...                                   ...  
8221500  35a81d0f6054b1ac873543fc52bdd8cf  
8221505  84b5da9f0c8e714e67021d5e880fc9d6  
8221506  1c921c208d7956f86e0c43be560ed030  
8221509  86d5addfa5f877eae6873a08743c4ea3  
8221510  c674d5089f9a277809f964b2855356fc  

[20633 rows x 6 columns]

 *De los 3329440 productos (obtenidos en el dataframe mercadona), hay un total de 20633 registros que se corresponden con Steinburg.*
 *Debemos tener en cuenta que  estamos viendo productos repetidos, por lo que debemos hallar valores únicos.*

In [28]:
uniqste = pd.DataFrame(blancaste['product_id'].unique())
uniqste

0
0   3c6d92df67a1e42852133818f6321b09
1   5d0c9d51dc38b9c139b22308cf50d1f8
2   2318cf2ee797dc035e4846a270ef9cf5
3   2e181ce72e732e6e18cdeef3be8921b3
4   d239934fa26e6a34473f1b2f324096a1
5   1152bc6fe534be109a7c02e7b3a93f26
6   93701986c3a5dc2bfe386c075b09301b
7   d9b54b063092fa4a8394816ea6af0d1e
8   6c90b85bfa2ee2288e142614c685c288
9   96c347a52dce17934da0b6ccec954023
10  f1bd0b03566e7027e0617fea5da067ab
11  c81ebb320092f51da60aaf0a7a1d0ff0
12  46758dc5bf8275a2968968377d291ebd
13  bbd065eaaf47f1ee4b9c03b5c64f8ebb
14  060caf4b2364457fbd87dc09428719cb
15  3a974af2f23c43d9739d4d0551ccce93
16  2594df47b786723d9e29fa33034762c1
17  82da9cc11a8e0fc9ea5a40a298083938
18  9840eeb21ae342314bc596cee186b6ca
19  994a70f164013dc13daedf9253fc4825
20  2c3d75bc48f4616404943cf5cad7aa47
21  6a1c02b9619543b935d6d293cf6ae2dc
22  35a81d0f6054b1ac873543fc52bdd8cf
23  8fc22beaebfd4a688ae7be721db1db0d
24  61b71df79b04e904c81f64a658682490
25  84b5da9f0c8e714e67021d5e880fc9d6
26  94382dc561fa8708b1b21dffa6d8fd30
27  ff657dd8827ddfb70899fdd5a10b4402
28  57aad976200aae0f8707ef3af2a4996a
29  f3a0f5a3d9e313f6040054ebc1588d14
30  86d5addfa5f877eae6873a08743c4ea3
31  1c2868fbe179c7df621d7a3c12f9c7d9
32  b39d95fbec81a5a453c7d6a14967df05
33  d716b8934a42264b583aff06f3f81dfe
34  ebf7038fcf45fdbac6e40832c1589177
35  370d0daf4a90f3340da3898775ec85b4
36  cfd15e749a7f35536d267cdf1a15bc68
37  2467764d098425beb82134afa417adc3
38  a814758dc6dc9c04ef749894def2b389
39  60070a770fd8019c3c1a1c3a50710ea0
40  ba7aa1d674d9a29a152342bbe78a10d2
41  21a7a46bd929d8fa090b7f0403864380
42  1c921c208d7956f86e0c43be560ed030
43  c674d5089f9a277809f964b2855356fc
44  ae08e4b7af6f049e270a18f6a9b2c24c
45  7f8ad65a2582a54a8302c46b5cf25f5b
46  b1f614a14b34b6552fd468125a89ffab

#### *Steinburg de Mercadona tiene 47 productos.* 

In [29]:
blancaste.describe()

price
count 20633.00
mean      1.84
std       1.62
min       0.22
25%       0.35
50%       0.95
75%       3.30
max       5.76

### Cálculo de Mediana con respecto al precio

In [30]:
#Mediana Día
print(blancadia['price'].median())  

1.49


In [31]:
#Mediana Carrefour
print(blancacar['price'].median())  

1.6


In [32]:
# Mediana Mercadona
print(blancahac['price'].median())   
print(blancabos['price'].median())   
print(blancadel['price'].median())   
print(blancaste['price'].median()) 

1.45
1.7
2.95
0.95


### Cálculo de Varianza con respecto al precio

In [33]:
#Varianza Día
print(blancadia['price'].var())  

3.6084005130582386


In [34]:
#Varianza Carrefour
print(blancacar['price'].var())   

31.925911103577672


In [35]:
# Varianza Mercadona
print(blancahac['price'].var())   
print(blancabos['price'].var())   
print(blancadel['price'].var())   
print(blancaste['price'].var())   

5.916744072148332
1.8783779557759477
3.639079445083436
2.612494787083848


### Conclusión

De acuerdo a la oferta que sostiene cada marca blanca de los supermercados, encontramos que:
* Supermercados Dia tiene 2414 productos distintos bajo su propia marca.
* Supermercados Carrefour cuenta con 3057 productos distintos bajo su propia marca.
* Supermercados Mercadona suma 4933 productos distintos bajo su propia marca, sólo que se agrupan bajo otros nombres:
    * Hacendado, relacionada con productos alimenticios, posee 2746 productos.
    * Bosque Verde, artículos relacionados con la limpieza hogareña, agrupa 532 productos.
    * Deliplus, relacionada con la higiene personal, tiene 1608 productos.
    * Steinburg, incluye cerveza de trigo, con 47 productos.

Así podemos sostener que, Mercadona es el supermercado que ofrece una mayor variedad de productos, en términos de cantidad y variedad. 

Además, en lo que respecta al precio de los productos de cada marca propia, a través del cálculo de la varianza del precio de los mismos, obtenemos que Carrefour es el que mayor variabilidad tiene, en segundo lugar Hacendado de Mercadona, en tercer lugar Deliplus de Mercadona y muy cerca Día.

Complementando este análisis, obtenemos que la mediana de mayor valor con respecto al precio de las marcas blancas de cada supermercado corresponde a Deliplus de Mercadona, en segundo lugar Bosque Verde y luego Carrefour.